File: frontend/division/seguridad/admin_general/tools/inject_ia.js
const fs = require('fs');
const path = require('path');

const vistasDir = 'vistas2';

if (!fs.existsSync(vistasDir)) {
    console.error('Directory not found:', vistasDir);
    process.exit(1);
}

const files = fs.readdirSync(vistasDir);

files.forEach(file => {
    if (!file.endsWith('.html') || !file.startsWith('vista')) return;

    const filePath = path.join(vistasDir, file);
    let content = fs.readFileSync(filePath, 'utf8');

    const match = file.match(/vista(\d+)/);
    if (!match) return;
    const vid = match[1];
    const spanId = `v${vid}_ia_analysis`;

    // Remove old block (aggressively to clean up old ones)
    // We remove any alert--info div that contains the specific id OR just any generic ia_analysis id to be safe
    // But be careful not to remove OTHER alert--infos.
    // We target the specific span id pattern.
    const oldBlockRegex = /<div class="alert alert--info"[\s\S]*?id="v\d+_ia_analysis"[\s\S]*?<\/div>/g;
    content = content.replace(oldBlockRegex, '');

    // Also remove any previous injection that might have used a different ID format if necessary
    // For now, assume consistent ID format v{N}_ia_analysis

    // New Block
    const newBlock = `
<div class="alert alert--info" style="margin-top: 1.5rem;">
    <strong>🤖 Análisis de Situación:</strong> <span id="${spanId}">Cargando análisis...</span>
</div>`;

    // Insert after question-section
    // We assume question-section ends with </div> and is followed by either <div class="cards-grid"> or similar.
    // We search for <div class="question-section" ...> ... </div>
    // To match until the closing div, we rely on indentation or structure.
    // The question-section usually ends line 21 in vista1.html.

    const qsStart = content.indexOf('<div class="question-section"');
    if (qsStart === -1) {
        console.warn(`Skipping ${file} (No question-section)`);
        return;
    }

    // Find the closing div of question-section.
    // Since question-section contains only h2 and p (no divs), the first </div> after qsStart is the closing one.
    const qsEnd = content.indexOf('</div>', qsStart);
    if (qsEnd === -1) {
        console.warn(`Skipping ${file} (No closing div for question-section)`);
        return;
    }

    // Insert after the closing div
    const before = content.substring(0, qsEnd + 6); // +6 for </div>
    const after = content.substring(qsEnd + 6);

    const newContent = before + '\n' + newBlock + after;

    fs.writeFileSync(filePath, newContent, 'utf8');
    console.log(`Updated ${file}`);
});

File: frontend/division/seguridad/admin_general/tools/inject_ia.py
import os
import re

vistas_dir = r"d:\GitHub\ALGARROBO_BASE\frontend\division\seguridad\admin_general\vistas2"

template = """
<div class="alert alert--info" style="margin-top: 1.5rem;">
    <strong>🤖 Análisis de Situación:</strong> <span id="{id}_ia_analysis">Analizando el comportamiento delictual de la semana...</span>
</div>
"""

for filename in os.listdir(vistas_dir):
    if filename.endswith(".html") and filename.startswith("vista"):
        filepath = os.path.join(vistas_dir, filename)
        try:
            with open(filepath, "r", encoding="utf-8") as f:
                content = f.read()
            
            # Determine vista ID
            match = re.search(r"vista(\d+)", filename)
            if not match: continue
            vid = match.group(1)
            span_id = f"v{vid}"
            
            # Remove existing IA block (aggressively to clean up old ones)
            content = re.sub(r'\s*<div class="alert alert--info"[\s\S]*?id="v\d+_ia_analysis"[\s\S]*?</div>', '', content)
            
            # Prepare new block
            new_block = template.format(id=span_id)
            
            # Insert after question-section
            # We assume question-section </div> closes the header block.
            # Regex: Find class="question-section", then find the next </div>
            # Since regex is greedy/lazy, we ensure we match until the first </div> after the opening tag.
            # But question-section has attributes.
            
            pattern = r'(<div class="question-section"[^>]*>[\s\S]*?</div>)'
            
            if re.search(pattern, content):
                new_content = re.sub(pattern, r'\1' + new_block, content, count=1)
                
                with open(filepath, "w", encoding="utf-8") as f:
                    f.write(new_content)
                print(f"Updated {filename}")
            else:
                print(f"Skipped {filename} (Pattern not found)")
        except Exception as e:
            print(f"Error processing {filename}: {e}")
